.github/workflows/build_and_test_cmake.yml

```diff
@@ -0,0 +1,441 @@
+name: Build and test w/ CMake
```

Comment:
CMake is missing in Cirrus CI which is covering a few flows that are not covered in GitHub Actions:
- FreeBSD
- Fuzzing tests (memory, address, undefined)

Reply:
I will do FreeBSD and introduce Sanitizer in another MR